dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeaccessionDatasetVersionCommand.java

```diff
@@ -10,35 +10,65 @@
 import edu.harvard.iq.dataverse.engine.command.CommandContext;
 import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
 import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.globalid.GlobalIdServiceBean;
+import edu.harvard.iq.dataverse.persistence.DvObject;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
 import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
 import edu.harvard.iq.dataverse.persistence.user.Permission;
+import io.vavr.control.Option;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
 
 /**
  * @author skraffmiller
  */
 @RequiredPermissions(Permission.PublishDataset)
 public class DeaccessionDatasetVersionCommand extends AbstractCommand<DatasetVersion> {
```

Comment:
Does `DeaccessionDatasetVersionCommand` was used anywhere before this MR? If not then I would move your changes to `DatasetDeaccessionService`. Long time ago we were trying to move away from the whole `CommandEngine` system (since commands are usually hard to test - or at least harder than it should be). After your changes we will check the permission twice and also put log entry in db twice. We rather want to avoid such situation

Reply:
Ok, that makes sense. I moved the code and removed `DeaccessionDatasetVersionCommand`. There was a small issue with static method `GlobalIdServiceBean.getBean` which uses the `CommandContext` to retrieve the necessary beans. I've refactored the code to a new bean `GlobalIdServiceBeanResolver` which does the look-up and marked the static method as deprecated.